m4/gx_mpi.m4

```diff
@@ -0,0 +1,141 @@
+# ----------------------------------------------------------------------
+# Check if the compiler can find the MPI headers and libraries
+AC_DEFUN([GX_MPI], [
+AC_PREREQ(2.50) dnl for AC_LANG_CASE
+
+AS_VAR_PUSHDEF([gx_cv_mpi_h],[gx_cv_[]_AC_LANG_ABBREV[]_mpi_h])
+dnl Check for the MPI header.  Here we use AC_TRY_COMPILE as AC_CHECK_HEADER
+dnl Note, we use AC_TRY_COMPILE as AC_CHECK_HEADER will call $CPP. Since
+dnl CC may be a mpi-specific compiler (e.g. mpicc), we don't want to use $CPP.
+AC_LANG_CASE([C], [
+    AC_REQUIRE([AC_PROG_CC])
+    AC_CACHE_CHECK([for mpi.h], [gx_cv_mpi_h], [dnl
+    gx_cv_mpi_h=no
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include <mpi.h>])], [dnl
+      gx_cv_mpi_h=yes])])
+],
+[C++], [
+    AC_REQUIRE([AC_PROG_CXX])
+    AC_CACHE_CHECK([for mpi.h], [gx_cv_mpi_h], [dnl
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include <mpi.h>])],[dnl
+      gx_cv_mpi_h=yes])])
+],
+[Fortran 77], [
+    AC_REQUIRE([AC_PROG_F77])
+    AC_CACHE_CHECK([for mpif.h], [gx_cv_mpi_h], [dnl
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [      include "mpif.h"])], [dnl
+      gx_cv_mpi_h=yes])])
+],
+[Fortran], [
+    AC_REQUIRE([AC_PROG_FC])
+    AC_CACHE_CHECK([for mpif.h], [gx_cv_mpi_h], [dnl
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [      include "mpif.h"])], [dnl
+      gx_cv_mpi_h=yes])])
+    GX_FC_CHECK_MOD([mpi])
+])
+# Check for library
+AC_SEARCH_LIBS([MPI_Init], [mpi])
+
+AS_VAR_PUSHDEF([gx_mpi_lang_usable], [gx_[]_AC_LANG_ABBREV[]mpi])
+AC_LANG_CASE([C], [test "$gx_cv_mpi_h" = yes -a "x$ac_cv_search_MPI_Init" != "xno" && gx_mpi_lang_usable=yes],
+    [C++], [test "$gx_cv_mpi_h" = yes -a "x$ac_cv_search_MPI_Init" != "xno" && gx_mpi_lang_usable=yes],
+    [Fortran 77], [test "$gx_cv_mpi_h" = yes -a "x$ac_cv_search_MPI_Init" != "xno" && gx_mpi_lang_usable=yes],
+    [Fortran], [test \( "$gx_cv_mpi_h" = yes -o $gx_cv_fc_check_mod_mpi = yes \) -a "x$ac_cv_search_MPI_Init" != "xno" && gx_mpi_lang_usable=yes
+])
+AS_VAR_IF([gx_mpi_lang_usable], [yes], [dnl
+    m4_default([$1], [])
+    AC_LANG_CASE([C], [AC_DEFINE([HAVE_MPI_H], 1, [Define to 1 if the mpi.h header file $1 is found])],
+        [C++], [AC_DEFINE([HAVE_MPI_H], 1, [Define to 1 if the mpi.h header file $1 is found])],
+        [Fortran 77], [AC_DEFINE([HAVE_MPI_H], 1, [Define to 1 if the mpif.h header file $1 is found])],
+        [Fortran 77], [AC_DEFINE([HAVE_MPI_H], 1, [Define to 1 if the mpif.h header file $1 is found])]
+    )], [dnl
+    m4_default([$2], [AC_MSG_ERROR([Unable to find the MPI headers or library for _AC_LANG])])
+])
+AS_VAR_POPDEF([gx_mpi_lang_usable])
+AS_VAR_POPDEF([gx_cv_mpi_h])
+])
+
+# GX_MPI_FC_LEGACY_INTERFACE()
+# ----------------------------------------------------------------------
+# Check if the Fortran MPI library uses legacy interfaces.
+# 
+# If the Fortran MPI library uses legacy interfaces, the variable
+# HAVE_MPI_FC_LEGACY will be set.
+AC_DEFUN([GX_MPI_FC_LEGACY_INTERFACE], [
+    AC_LANG_ASSERT([Fortran])
```

Comment:
This check was always failing during my testing on the amd box. It ended up being the extension of the file during the compilation test. I added `AC_FC_SRCEXT([F90])` to here and it fixed it, not sure if there's a better solution.